extensions/Gemini/python/src/aiconfig_extension_gemini/Gemini.py

```diff
@@ -134,17 +135,36 @@ async def deserialize(
                 "on_deserialize_start", __name__, {"prompt": prompt, "params": params}
             )
         )
-        resolved_prompt = resolve_prompt(prompt, params, aiconfig)
 
         # Build Completion data
         model_settings = self.get_model_settings(prompt, aiconfig)
 
         completion_data = refine_chat_completion_params(model_settings)
 
-        messages = self._construct_chat_history(prompt, aiconfig, params)
-        messages.append({"role": "user", "parts": [{"text": resolved_prompt}]})
+        if contains_prompt_template(prompt):
+            messages = self._construct_chat_history(prompt, aiconfig, params)
 
-        completion_data["contents"] = messages
+            resolved_prompt = resolve_prompt(prompt, params, aiconfig)
+            
+            messages.append({"role": "user", "parts": [{"text": resolved_prompt}]})
+            
+            completion_data["contents"] = messages
+        else:
+            # If contents is already set, do not construct chat history. TODO: @Ankush-lastmile
```

Comment:
If contents is already set (ie: defined as a key in a dict --> see 3rd and 4th case in "supported types" below)